src/components/converter/Form.js

```diff
@@ -127,8 +128,26 @@ function useConvertInputs(symbol, tokenToAnjRate, anjToTokenRate) {
       if (tokenDecimals === -1 || anjDecimals === -1) {
         return
       }
-
-      const properTokenRate = bigNum(toWei(tokenToAnjRate.toString()))
+      let value = '0'
+      if (event.target.value && symbol === 'USDC') {
+        value = new BigNumber(event.target.value)
+          .multipliedBy(10 ** 6)
+          .toFixed(0, 1)
+      } else if (event.target.value) {
+        value = new BigNumber(event.target.value)
+          .multipliedBy(10 ** 18)
+          .toFixed(0, 1)
+      }
+      const executionRate = getTradeDetails(
+        TRADE_EXACT.INPUT,
+        value,
```

Comment:
`value` here and in `handleAnjChange()` can be `NaN` if the user accidentally inputs a non numeric string (e.g. `10.12a`)

Reply:
Fixed in #62